maven/lib/dependabot/maven/file_parser.rb

```diff
@@ -123,6 +120,10 @@ def build_dependency(pom, dependency_node, name)
             }.merge(property_details)
           }]
         )
+
+        classifier = dependency_classifier(dependency_node, pom)
+        dependency.requirements.first[:metadata][:classifier] = classifier if classifier
+        dependency
```

Comment:
🔎 What do you think of doing this inline similarly to `groups: dependency_groups(pom, dependency_node)`? That'd look something like:

```ruby
        Dependency.new(
          name: name,
          version: dependency_version(pom, dependency_node),
          package_manager: "maven",
          requirements: [{
            requirement: dependency_requirement(pom, dependency_node),
            file: pom.name,
            groups: dependency_groups(pom, dependency_node),
            source: nil,
            metadata: {
              packaging_type: packaging_type(pom, dependency_node)
              classifier: dependency_classifier(dependency_node, pom)
            }.merge(property_details).compact
          }]
        )
```

(with an added `Hash#compact` to take care of the `nil` case)

My thinking is that we can rely on the implicit return value and conform the surrounding style (for `version`, `requirements` and `groups`)

Reply:
Ah, I had started doing this but the `nil` was failing tests. Good idea adding `compact`!